### DISCUSSION TITLE
After oauth2 I want to check if user already exist

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#5111] [Q&A] [answered]
Hey,

I have the use case where a user can register/login via oauth2 but I want that new users should have "freeTrial" === true.

Not sure how to check if the oauth2 user who just logged in was already created in the database or not

```
let newUser = await locals.pb?.collection('users')
            .authWithOAuth2Code(provider.name, code, expectedVerifier, redirectURL);
        
        await locals.pb?.collection('users').update(
            newUser['record']['id'], {
                'freeTrial' : true
            });
```

**Comment by ganigeorgiev** [comment, marked as answer]
There should be `newUser.meta.isNew` property indicating whether the account was created by the OAuth2 auth.

Alternatively, you can also set the `freeTrial:true` as part of the optional createData argument:

```js
let newUser = await locals.pb?.collection('users')
            .authWithOAuth2Code(provider.name, code, expectedVerifier, redirectURL, { freeTrial: true });
```

**Reply by ganigeorgiev** [reply]
Another (maybe easier) alternative could be to negate the field, aka. eg. `nonFreeTrial/paid/upgraded/etc.`, which will make the default value (`false`) automatically as free trial for new users.

**Reply by MuslemRahimi** [reply]
newUser.meta.isNew works great for me. Thanks!

**Reply by MuslemRahimi** [reply]
> There should be `newUser.meta.isNew` property indicating whether the account was created by the OAuth2 auth.
> 
> Alternatively, you can also set the `freeTrial:true` as part of the optional createData argument:
> 
> ```js
> let newUser = await locals.pb?.collection('users')
>             .authWithOAuth2Code(provider.name, code, expectedVerifier, redirectURL, { freeTrial: true });
> ```

Is there maybe a way to set the freeTrial to true if the user verified the email address?

**Reply by ganigeorgiev** [reply]
If the PocketBase user was created by the OAuth2 provider it will be always `verified`.

For other use cases you'll have to use the hooks but without details about your use case I don't understand the full picture and I don't think I can help you further.

**Reply by MuslemRahimi** [reply]
> If the PocketBase user was created by the OAuth2 provider it will be always `verified`.
> 
> For other use cases you'll have to use the hooks but without details about your use case I don't understand the full picture and I don't think I can help you further.

Sorry I forgot to add the code for manual registration

```

		try {
			const newUser = await locals.pb.collection('users').create(formData);
			await locals.pb?.collection('users').update(
				newUser?.id, {
					'freeTrial' : true,
					'tier': 'Pro', //Give new users a free trial for the Pro Subscription
			});

			await locals.pb.collection('users').requestVerification(formData.email);
		} catch (err) {
			console.log('Error: ', err);
			error(err.status, err.message);
		}
```
I would only give a user a free trial if the email was verified first. Im not sure how to do this in pockebase.
